### ISSUE TITLE
Sending a "quick reply" in a notification leads to clinging "unread" messages in the room

### ISSUE THREAD

**Issue by Ezwen** [vector-im/riotX-android#550]
Each time I use the "quick reply" action from within a RiotX Android notification, the following happens:
- my answer is sent successfully,
- the corresponding room in RiotX shows an "unread message" (probably the message I answered to − maybe already at this point this is a bug),
- if I open the room in RiotX, the "unread message" counter remains, as if opening (and scrolling through) the history in the said room does not manage to make the new message count as read,
- if I open the same room in Riot web, the message is finally registered as read, and the "unread message" counter disappears from the room in RiotX as well.

In other words, it seems there is a bug in tagging messages as read/unread when the quick reply action is used?

**Comment by auscompgeek** [comment]
It looks like the unread counter is of the previously notified messages, rather than the reply the user sent.

**Comment by rajil** [comment]
Can confirm the bug on Beta Riot 0.91.3 [400091032] (G-b3120).

**Comment by 532910** [comment]
> Each time I use the "quick reply" action from within a RiotX Android notification, the following happens:

1. I see endless rotation animation

![image](https://user-images.githubusercontent.com/14617699/87091195-efcbe000-c241-11ea-80a1-337af789ccde.png)


2. If I press this notification I got my message in a new notification:
![image](https://user-images.githubusercontent.com/14617699/87091282-17bb4380-c242-11ea-9d7b-b74e017ff495.png)

**Comment by 532910** [comment]
I believe this issue must be riot-go-live, mustn't it?

**Comment by T-bond** [comment]
If I open the room, go back in history and scroll back to the newest message, the undread message indicator dissaperas correctly. (So no need for Element Web)

The other problems still apply, unfortunately (spinning indicator not vanishing, message won't get marked as read)

1.0.11 [40100110] (F-dffdcfe1)

**Comment by immanuelfodor** [comment]
I can confirm this, it's still an issue. Also reported in https://github.com/vector-im/element-android/issues/1139

**Comment by immanuelfodor** [comment]
This and the infinite quick reply spinner are the most annoying bugs of Element for me, experiencing it daily :(

**Comment by chagai95** [comment]
I think replying implies also marking as read, maybe the original idea was to have them somehow separated? I doubt it, but I can't think of another reason. Maybe the idea of not clearing was to have the option to reply with multiple separate messages from these action buttons?

**Comment by chagai95** [comment]
I tried playing around with it a bit now, I think this stuff is extremely difficult to debug, it seems like I keep on running into unrelated issues, the last issue I ran into is that on the dbg version the mark read does not work all together. so I'm just gonna try to release this [commit ](https://github.com/chagai95/element-android/commit/5b95316cfd9f36d8912c82b66bb4a70598cd4bd7) as is and see if it's better like this, you're welcome to try it as well if you like and hopefully soon I can create a PR if I see it works.

**Comment by chagai95** [comment]
I released my changes on [this app](https://play.google.com/store/apps/details?id=matrix.chagai.website) on the Playstore if anyone wants to try them out, I think the behavior is now a bit better but not perfect, I'll be testing this in the next few weeks and if I'm happy I'll create a PR
